src/java.base/windows/classes/sun/nio/fs/WindowsLinkSupport.java

```diff
@@ -190,20 +190,92 @@ static String getRealPath(WindowsPath input, boolean resolveLinks)
         WindowsFileSystem fs = input.getFileSystem();
 
         // Start with absolute path
+        WindowsPath absolute = null;
         String path = null;
         try {
-            path = input.toAbsolutePath().toString();
+            absolute = input.toAbsolutePath();
+            path = absolute.toString();
         } catch (IOError x) {
             throw (IOException)(x.getCause());
         }
 
         // Collapse "." and ".."
         if (path.indexOf('.') >= 0) {
-            try {
-                path = GetFullPathName(path);
-            } catch (WindowsException x) {
-                x.rethrowAsIOException(input);
+            // The Windows function GetFullPathName() does not access the
+            // file system, hence is susceptible to removing symbolic links
+            // which are followed by the special name ".." whether or not
+            // links are being followed. Examine the path element-by-element
+            // to determine whether this would occur, and if so use the
+            // full path manually derived in the process.
+            String fullpath = null;
+            if (path.contains("..")) {
```

Comment:
I suggest to extract code for this case to a separate method. `sun.nio.fs.WindowsLinkSupport#getRealPath` is already too big.

Reply:
Sounds like a good idea.